src/pygama/dsp/processors/histogram.py

```diff
@@ -0,0 +1,145 @@
+from __future__ import annotations
+
+from math import floor
+
+import numpy as np
+from numba import guvectorize
+
+from pygama.dsp.utils import numba_defaults_kwargs as nb_kwargs
+
+
+@guvectorize(
+    [
+        "void(float32[:], float32[:], float32[:])",
+        "void(float64[:], float64[:], float64[:])",
+    ],
+    "(n),(m),(m)",
```

Comment:
@iguinn `borders_out` is actually one element longer than `weights_out`. How should the signature look like here?